book/src/design/keys.md

```diff
@@ -0,0 +1,86 @@
+  also removes an issue where two full viewing keys could be constructed that have the
+  same $\mathsf{ivk}$ but different $\mathsf{ovk}$s. Users still have control over whether
+  $\mathsf{ovk}$ is used when constructing a transaction.
+
+- All diversifiers now result in valid payment addresses, due to group hashing into Pallas
+  being specified to be infallible. This removes significant complexity from the use cases
+  for diversified addresses.
+
+Other than the above, Orchard retains the same design rationale for its keys and addresses
+as Sapling. For example, diversifiers remain at 11 bytes, so that Orchard addresses are
+the same length as Sapling addresses.
+
+## Hierarchical deterministic wallets
+
+When designing Sapling, we defined a [BIP 32]-like mechanism for generating hierarchical
+deterministic wallets in [ZIP 32]. We decided at the time to stick closely to the design
+of BIP 32, on the assumption that there were Bitcoin use cases that used both hardened and
+non-hardened derivation that we might not be aware of. This decision created significant
+complexity for Sapling: we needed to handle derivation separately for each component of
+the expanded spending key and full viewing key (whereas for transparent addresses there is
+only a single component in the spending key).
+
+Non-hardened derivation enables creating a multi-level path of child addresses below some
+parent address, without involving the parent spending key. The primary use case for this
+is HD wallets for transparent addresses, which use the following structure defined in
+[BIP 44]:
+
+- (H) BIP 44
+  - (H) Coin type: Zcash
+    - (H) Account 0
+      - (N) Normal addresses
+        - (N) Address 0
+        - (N) Address 1...
+      - (N) Change addresses
+        - (N) Change address 0
+        - (N) Change address 1...
+    - (H) Account 1...
+
+Shielded accounts do not require separating change addresses from normal addresses, due to
+addresses not being revealed in transactions. Similarly, there is also no need to generate
+a fresh spending key for every transaction, and in fact this causes a linear slow-down in
+wallet scanning. But for users that do want to generate multiple addresses per account,
+they can generate the following structure, which does not use non-hardened derivation:
+
+- (H) ZIP 32
+  - (H) Coin type: Zcash
+    - (H) Account 0
+      - Diversified address 0
+      - Diversified address 1...
+    - (H) Account 1...
+
+Non-hardened derivation is therefore only required for use-cases that require the ability
+to derive more than one child layer of addresses. However, in the years since Sapling was
+deployed, we have not seen *any* such use cases appear.
+
+Therefore, for Orchard we only define hardened derivation, and do so with a much simpler
+design that ZIP 32. All derivations produce an opaque binary spending key, from which the
+keys and addresses are then derived.
```

Comment:
```suggestion
keys and addresses are then derived. As a side benefit, this makes key formats
shorter. (The formats that will actually be used in practice for Orchard will correspond
to the simpler Sapling formats in the protocol specification, rather than the longer
and more complicated "extended" ones defined by ZIP 32.)
```

Reply:
This might not quite be the case (we might still want something akin to the "chain code" in ZIP 32 when actually deriving spending keys), but the spending key structure itself should end up the same. We should decide how to derive a diversifier key though (probably just from the spending key bytes, rather than as an artifact of HD derivation).

Reply:
We should pair on working out how the ZIP 32 derivation will work for Orchard.